### ISSUE TITLE
Question: is there a way of changing the state of a download?

### ISSUE THREAD

**Issue by kevinpawsey** [DobyTang/LazyLibrarian#1467]
To help with identifying and fixing issues, please include as much information as possible, including:
### LazyLibrarian version number (at the bottom of config page)
Current Version: d54b8d1cf04ed2154cd34d2e3e36ef468b6ce7a8 : Latest Version: d54b8d1cf04ed2154cd34d2e3e36ef468b6ce7a8

### Operating system used (windows, mac, linux, NAS type)
Ubuntu 17.10 up to date
### Interface in use (default, bookstrap)
Bookstrap
### Which api (Goodreads, GoogleBooks, both)
both
### Source of your LazyLibrarian installation (git, zip file, 3rd party package)
git

This isn't so much of an error (I don't think), as a bugette somewhere... but it appears that now I have managed to sort my source out for downloading audiobooks, it has successfully grabbed the books (yey!)... and downloaded them.  The trouble is, on completion it seems to have marked them as "Aborted" instead of completed.

I have checked in deluge, and the downloads are definitely ok, and complete... sitting at 100% and seeding.  I have also tried to manually run the post processing (Audiobooks/Run Post-Processor), but I get this in the log:
`2018-06-15 13:54:04 | ERROR | Unhandled exception in processDir: Traceback (most recent call last): File "/home/plex/LazyLibrarian/LazyLibrarian/lazylibrarian/postprocess.py", line 888, in processDir myDB.action(q, (book['NZBurl'])) File "/home/plex/LazyLibrarian/LazyLibrarian/lazylibrarian/database.py", line 48, in action sqlResult = self.connection.execute(query, args) ProgrammingError: Incorrect number of bindings supplied. The current statement uses 1, and there are 656 supplied.
-- | -- | --
2018-06-15 13:54:04 | ERROR | Fatal error executing UPDATE wanted SET Status="Failed" WHERE NZBurl=? and Status="ed" :: Incorrect number of bindings supplied. The current statement uses 1, and there are 656 supplied.
2018-06-15 13:54:04 | WARNING | A Song of Ice and Fire 2 - A Clash of Kings was aborted by delugerpc, deleting failed task
2018-06-15 13:54:04 | INFO | 0 books/mags processed.`
I've also tried restarting LL, but I haven't tried restarting deluge (everything else seems to be working ok on deluge).

If you need any other information, just let me know.

**Comment by philborman** [comment]
The answer to the question is Yes, but only if it's Processed or Failed.

Two things here...
1. The update error is a typo, fixed now.
2. We think delugerpc aborted the downloads 
`WARNING | A Song of Ice and Fire 2 - A Clash of Kings was aborted by delugerpc`
so why do we think that?   With the typo fixed the "Aborted" should get changed to "Failed" and you will be able to click on the "Failed" button to get the reason for the failure and change the status or delete.
Before that, if you are interested, we can interrogate the database directly from the commandline, would be interesting to see what the problem was. Could be we are misinterpreting the progress result.
In a terminal
```
sqlite3 /path/to/lazylibrarian.db
SELECT DLResult from wanted where Status="Aborted";
```
Note the semicolon on the end of the SELECT statement. To exit sqlite use   
`.quit  ` (note the period at the start and no semicolon on the end)
As a work-around for the postprocessing problem, once you have done this and got the DLResult messages,
STOP deluge completely (so when we ask for progress it can't reply) and back in sqlite3
`UPDATE wanted set Status="Snatched" where Status="Aborted";`
Now run the postprocessor and it will notice all the "Snatched" books have suddenly appeared in the download folder. It only looks for Snatched ones, not Aborted.

**Comment by kevinpawsey** [comment]
Here is the results of the query, as requested:
Firstly I stopped LazyLibrarian (to make sure that there were no database changes whilst doing the query)
`plex@plex-mc:~/LazyLibrarian/LazyLibrarian$ sqlite3 lazylibrarian.db
SQLite version 3.19.3 2017-06-08 14:26:16
Enter ".help" for usage hints.
sqlite> SELECT DLResult from wanted where Status="Aborted";
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
OK
sqlite> `

Stopped both deluged and deluge-web
Ran the query to update the status to "Snatched" as requested.
Started LL again.
Run the post-process and it is now processing everything properly.

Thank you for your help... is the spelling mistake corrected in a new commit?
